src/engine/controls/loopingcontrol.cpp

```diff
@@ -1587,6 +1597,18 @@ void LoopingControl::slotLoopMove(double beats) {
     }
 }
 
+// Must be called from the engine thread only
+mixxx::audio::FramePos LoopingControl::seekInsideCurrentLoop(
+        mixxx::audio::FramePos currentPosition) {
+    if (!m_bLoopingEnabled) {
+        return currentPosition;
+    }
+    LoopInfo loopInfo = m_loopInfo.getValue();
+    return seekInsideAdjustedLoop(currentPosition,
```

Comment:
> If I return here loopInfo.endPosition, the loop is left after disable slip.
...
A bug in the first conditions in seekInsideAdjustedLoop.

True. I fixed that.  Though, that only affected the new slip looping. All existing function (EngineBuffer, ReadAheadManager) already handle that appropriately.

> This is because the end position is the first position after the loop.

I don't understand. If [`ReadAheadManager::getNextSamples`](https://github.com/mixxxdj/mixxx/blob/de9a64defa5d7a7e6b02ccac2aebc4fc6820b942/src/engine/readaheadmanager.cpp#LL51C1-L80C6) is exactly at the trigger point (loop_out, or _in if reverse) it'll consider that as "inside the loop, trigger reached" and wrap around.